### PR TITLE
Upgrade OTel dependencies to v0.42.0.

### PR DESCRIPTION
**Pull request by jefchien**
**Description:** Re-defined `ExpandConverter` for environment variable expansion in config provider. Fixed tests that broke due to change.

**Link to tracking Issue:** https://github.com/aws-observability/aws-otel-collector/issues/867

**Testing:** Ran unit tests and built without issue.

### CODE REVIEW

**Review by Aneurysm9 — APPROVED** [review]

**Review by khanhntd — APPROVED** [review]
Nice catch on license. LGTM

File: pkg/config/config_factory_test.go [hunk 1]
```diff
@@ -102,21 +100,18 @@ func TestGetMapProviderContainer(t *testing.T) {
...
 	defer os.Unsetenv("PPROF_ENDPOINT")
 
-	os.Setenv("AOT_CONFIG_CONTENT", "extensions:\n  health_check:\n  pprof:\n    endpoint: '${PPROF_ENDPOINT}'\nreceivers:\n  otlp:\n    protocols:\n      grpc:\n        endpoint: 0.0.0.0:4317\nprocessors:\n  batch:\nexporters:\n  logging:\n    loglevel: debug\n  awsxray:\n    local_mode: true\n    region: 'us-west-2'\n  awsemf:\n    region: 'us-west-2'\nservice:\n  pipelines:\n    traces:\n      receivers: [prometheusreceiver]\n      exporters: [logging,awsxray]\n    metrics:\n      receivers: [prometheusreceiver]\n      exporters: [awsemf]\n  extensions: [pprof]")
-	defer os.Unsetenv("AOT_CONFIG_CONTENT")
+	os.Setenv(envKey, "extensions:\n  health_check:\n  pprof:\n    endpoint: '${PPROF_ENDPOINT}'\nreceivers:\n  otlp:\n    protocols:\n      grpc:\n        endpoint: 0.0.0.0:4317\nprocessors:\n  batch:\nexporters:\n  logging:\n    loglevel: debug\n  awsxray:\n    local_mode: true\n    region: 'us-west-2'\n  awsemf:\n    region: 'us-west-2'\nservice:\n  pipelines:\n    traces:\n      receivers: [otlp]\n      exporters: [logging,awsxray]\n    metrics:\n      receivers: [otlp]\n      exporters: [awsemf]\n  extensions: [pprof]")
```
Comment (sethAmazon) [inline pkg/config/config_factory_test.go:103]: good catch?

**Review by sethAmazon — APPROVED** [review]

**Comment by jefchien** [comment]
@Aneurysm9 Do we want to pull in the changes to flags.go (https://github.com/open-telemetry/opentelemetry-collector/blob/v0.42.0/service/flags.go)? Right now, we only allow the config/set flags. Pulling in this change would include 3 additional flags (feature-gates, metrics-addr, metrics-level).

**Comment by Aneurysm9** [comment]
`metrics-addr` and `metrics-level` are deprecated and can be set through configuration, so we don't want to reintroduce them.  We should bring in support for `feature-gates` though.

**Review by Aneurysm9 — APPROVED** [review]

**Pull request merged**